alws/build_planner.py

```diff
@@ -159,29 +163,64 @@ async def add_task(self, task: build_schema.BuildTaskRef):
             raw_refs, module_templates = await build_schema.get_module_refs(
                 task, self._platforms[0]
             )
-        refs = [
-            models.BuildTaskRef(
-                url=ref.url,
-                git_ref=ref.git_ref,
-                ref_type=BuildTaskRefType.GIT_BRANCH
-            ) for ref in raw_refs
-        ]
         module = None
         if self._build.mock_options:
             mock_options = self._build.mock_options.copy()
             if not mock_options.get('definitions'):
                 mock_options['definitions'] = {}
         else:
             mock_options = {'definitions': {}}
+        beholder_client = BeholderClient(
+            host=settings.beholder_host,
+            token=settings.beholder_token,
+        )
+        modules_to_exclude = []
         for platform in self._platforms:
             modularity_version = platform.modularity['versions'][-1]
+            clean_dist_name = re.search(
+                r'(?P<dist_name>[a-z]+)', platform.name, re.IGNORECASE,
+            ).groupdict().get('dist_name', '')
+            distr_ver = platform.distr_version
             if task.module_platform_version:
                 modularity_version = next(
                     item for item in platform.modularity['versions']
                     if item['name'] == task.module_platform_version
                 )
             for arch in self._request_platforms[platform.name]:
                 module = ModuleWrapper.from_template(module_templates[0])
+                endpoint = (
+                    f'/api/v1/distros/{clean_dist_name}/{distr_ver}'
+                    f'/module/{module.name}/{module.stream}/{arch}/'
+                )
+                beholder_response = None
+                pkgs_to_add = []
+                if not self._skip_module_checking:
+                    try:
+                        beholder_response = await beholder_client.get(endpoint)
+                    except Exception:
+                        logging.error('Cannot get module info')
```

Comment:
You can add `continue` here and remove `if beholder_response is not None:`

Reply:
Fixed